zeebe/dynamic-config/src/main/java/io/camunda/zeebe/dynamic/config/RoutingStateInitializer.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.dynamic.config;
+
+import io.camunda.zeebe.dynamic.config.state.ClusterConfiguration;
+import io.camunda.zeebe.dynamic.config.state.RoutingState;
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+import io.camunda.zeebe.scheduler.future.CompletableActorFuture;
+import java.util.Optional;
+
+/**
+ * Initializes the routing state of the cluster configuration if the partition scaling feature is
+ * enabled. All members will initialize the same routing state (as long as their statically
+ * configured partition counts match).
+ */
+public class RoutingStateInitializer implements ClusterConfigurationModifier {
+
+  private final boolean enablePartitionScaling;
+  private final int staticPartitionCount;
+
+  public RoutingStateInitializer(
+      final boolean enablePartitionScaling, final int staticPartitionCount) {
+    this.enablePartitionScaling = enablePartitionScaling;
+    this.staticPartitionCount = staticPartitionCount;
+  }
+
+  @Override
+  public ActorFuture<ClusterConfiguration> modify(final ClusterConfiguration configuration) {
+    if (configuration.routingState().isPresent() || !enablePartitionScaling) {
+      return CompletableActorFuture.completed(configuration);
```

Comment:
:thought_balloon: Does using `CompletableActorFuture` causes any problems in tests? The other classes uses the provided `ConcurrencyControl` to create the future, so that in tests we can use `TestConcurrencyControl` and `TestActorFuture`. Probably not an issue any more because now CompletableActorFuture can deal with non-actors in most cases.

Reply:
Ah, I was wondering why we were using the executor to create futures in the other initializer :+1:
It hasn't caused any issues yet so I'm inclined to leave it as is.